src/common.cpp

```diff
@@ -1874,13 +1880,14 @@ size_t buff_size = sizeof buff;
 #else
     int name[] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
 #endif
-    int result = sysctl(name, sizeof(name) / sizeof(int), buff, &buff_size, nullptr, 0);
-    if (result != 0) {
+    int ret = sysctl(name, sizeof(name) / sizeof(int), buff, &buff_size, nullptr, 0);
+    if (ret != 0) {
         wperror(L"sysctl KERN_PROC_PATHNAME");
     } else {
-        return std::string(buff);
+        buffstr = buff;
+        return result;
```

Comment:
This sets buffstr and returns result. That means you'll get a wrong answer and it'll be computed every time.

Just set result directly, no need for buffstr here.